agent/agentbootstrap/bootstrap.go

```diff
@@ -279,6 +284,54 @@ func verifyModelConfigDefaultSpace(st *state.State) error {
 	return errors.Annotatef(err, "cannot verify %s", config.DefaultSpace)
 }
 
+func validateCAASImageRepo(cfg controller.Config) error {
```

Comment:
This should be done in the `juju bootstrap` command, as the CAASImageRepo can be a PATH on the 🤬 client.

See [this in my WIP pr](https://github.com/juju/juju/blob/e8bc592dc73e2e6a1de63cda803025e37f9055b6/cmd/juju/commands/bootstrap.go#L1501)

Reply:
We resolve the details there then its fine to be a string in controllerconfig.